### DISCUSSION TITLE
Search Settings Feature: Feedback & Suggestions

### DISCUSSION THREAD

**Discussion by confused-Techie** [pulsar-edit/.github#150] [Core Application & Packages]
With the new version of Pulsar we have an exciting new feature, the ability to search through all settings of Pulsar in one place!

This new feature is still undergoing development, and as of now the search results wouldn't be considered final.

This discussion post can be a way to continue in depth discussion about the search results, how they can be improved, where they fail, and any other manner of concerns about them. 

So please feel free to drop a comment about what they do well, or could aim to do better!

**Comment by Spiker985** [comment]
I've encountered 2 things while using the feature:
1) My base instinct it to click on the entire result card, not the title - so I keep having to click, and then click again as I remember
2) Toggling the `Settings Icon in Status Bar` doesn't immediately remove the icon, it requires a restart

Less about this feature and more towards the fact of this package in general - we may wish to have toggles either inside settings-view, or adjust ppm, but there should be a toggle as to whether or not disabled packages get checked when checking for updates

**Reply by confused-Techie** [reply]
@Spiker985 Do you know if any other icons within the status bar are removed as soon as you change the config?
I haven't seen it myself, but don't change those settings often. But I'd assume we could set up a watcher on that config option, and change as needed.

And we could rather easily add a click handler to the whole package card. It was done the way it is currently to try and match as closely as possible to the behavior of the package cards, which the design is based off of. But appreciate the feedback!

**Reply by Spiker985** [reply]
I actually don't know if the status bar icon piece is possible or not, though I'm unsure why it wouldn't be

Just bringing to mind something I've seen

You're quite welcome. Love having the feature outright though

**Reply by Spiker985** [reply]
@confused-Techie How difficult would it be to have a setting to toggle "entire card navigation" versus "card title navigation"? :thinking:

**Reply by confused-Techie** [reply]
Do you mean to be able to change the setting of each item within the search settings? @Spiker985

**Reply by Spiker985** [reply]
On a slightly different note, I can confirm that disabling the GitHub package outright, does immediately get rid of the github status bar entry

Same with busy-signal

**Reply by Spiker985** [reply]
> Do you mean to be able to change the setting of each item within the search settings? @Spiker985

![image](https://github.com/pulsar-edit/.github/assets/7829451/0c1815cf-1f2b-446b-a959-3ff53bce3d62)
Kinda like this

**Reply by confused-Techie** [reply]
@Spiker985 And just to be clear, by navigator you mean that when you click the entire box it takes you to that place in settings?

Tbh I think originally it was like that, but I don't think it'd be too hard, I'll try to remember that one, but feel free to ping me if I forget or make a feature request. Since it's a good idea

**Reply by Spiker985** [reply]
I remember that being your default implementation, yeah.

There was discussion back and forth regarding it, and it settled on the title - which keeps messing me up. So I figure, if it's not too hard, than we give the choice for the user

**Reply by confused-Techie** [reply]
Yeah totally agree, it'd be the way to go, and you know how much I love adding endless configuration options lol

**Comment by ronherrema** [comment]
I would like to be able to search only within the Editor settings, but there doesn't seem to be a way to do that.

**Reply by Daeraxa** [reply]
Thats a good point, I wonder if we could introduce a special syntax in the search for that?

For example maybe an `@` syntax to filter, e.g. `@editor soft tabs` or `@packages project`.

**Comment by savetheclocktower** [comment]
Used this feature today because I found a page somewhere saying there was a “Scroll Past End” setting in the Editor. Instead of hunting for it, I thought I'd use search. I typed in `scroll past end` exactly.

It found it, but it was the eighth result:

<p><img width="628" alt="Screenshot 2025-06-19 at 9 56 57 AM" src="https://github.com/user-attachments/assets/0deb59e7-715b-4705-acd4-6042a7ecf03f" /></p>

I think the search algorithm might need some tweaking. It's tough because in some ways we want it to act like fuzzy search (like in the palettes), but in other ways we want it to act like more traditional search (e.g., typing `foo bar` should reveal a setting named “Bar Foo” even though the search terms weren't in the right order). Maybe there's a way to combine those approaches?
